### DISCUSSION TITLE
Documentation of Components Section

### DISCUSSION THREAD

**Discussion by dtekinoglu** [networkx/networkx#5464] [Ideas]
In the documentation of number_connected_component method, field "Raises: NetworkXNotImplemented" is not stated. However, the function actually raises NetworkXNotImplemented if the graph is directed. For other functions in the components section,  this part is included in the documentation. In order to provide consistency in documentations, I believe it can be helpful to include "Raises: NetworkXNotImplemented" also in number_connected_component page.

Besides, if a function in the components section only works for directed graphs (or only for undirected graphs), we can state this in the short descriptions given in the list. (See: https://networkx.org/documentation/stable/reference/algorithms/component.html)

For example; in the current version, short description of number_strongly_connected_components is as follows:

_"Returns number of strongly connected components in graph."_

We can make the following addition so that it becomes easier for user to choose which functions to use without clicking on them for more detail:

_"Returns number of strongly connected components in _**directed**_ graph."_

@MridulS @dschult if you think these would be helpful, I can gladly work on them.

**Comment by dschult** [comment]
That seems like a good idea!

**Comment by rossbar** [comment]
> In the documentation of number_connected_component method, field "Raises: NetworkXNotImplemented" is not stated.

I'm a bit confused by this - the `Raises` field in the webpage linked in the OP does explicitly state that `NetworkXNotImplemented` exceptions are raised when the graph is undirected.

> We can make the following addition so that it becomes easier for user to choose which functions to use without clicking on them for more detail

Indeed this seems like an improvement :+1:

**Reply by dtekinoglu** [reply]
@rossbar I still cannot see Raises field in the following link: https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.components.number_connected_components.html#networkx.algorithms.components.number_connected_components
Am I checking the wrong web page? Many thanks

**Reply by rossbar** [reply]
Oops, my mistake - you're right, there is no `Raises` section for the `number_connected_components` function. I must've clicked on one of the other functions instead - my mistake!

**Reply by dtekinoglu** [reply]
@rossbar Which file should I work on to modify general table of Components algorithms ([See this page](https://networkx.org/documentation/stable/reference/algorithms/component.html)) ? I've found connected.py to add `Raises`  for `number_of_connected_components` however I could not find the file for the table.

**Reply by MridulS** [reply]
@dtekinoglu That page is generated via https://github.com/networkx/networkx/blob/main/doc/reference/algorithms/component.rst and if you want to change the text next to the link to the function, for example the text `Returns the number of connected components."` next to `number_connected_components`
<img width="787" alt="Screenshot 2022-04-05 at 18 22 10" src="https://user-images.githubusercontent.com/5363860/161775641-4e4e4493-d145-4b86-b75f-2db6a89ef541.png">
 
you need to go to the function docstring and change the first line.
https://github.com/networkx/networkx/blob/fe23b915b4990d21050b5849e082f5da21b6c3c9/networkx/algorithms/components/connected.py#L69-L72

The documentation pipeline will pick up the first sentence, until it catches a new line.
